docs/source/examples/pooling-recycled.py

```diff
@@ -0,0 +1,78 @@
+from collections import defaultdict
+
+import asyncio
+import aio_pika
+from aio_pika.pool import Pool
+
+
+class NonRestoringRobustChannel(aio_pika.RobustChannel):
+    """
+    Custom robust channel that when reopened, does not restore any
+    queues or exchanges.
+    """
+    async def reopen(self) -> None:
+        # Clear out exchanges and queues when reopened
+        self._exchanges = defaultdict(set)
+        self._queues = defaultdict(set)
```

Comment:
By the way, we had some memory leaks with `set` so we replaced it by `WeakSet` in the `__Init__` method and now it's ok 🙃

Reply:
Thanks for the info! I will make those same changes...